# Natnicha Palakawong na ayuthaya

# <font color='red'> Analysis of shoppers to help Amazon reach its target market efficiently

In [1]:
features <- read.csv('/Users/Natnicha/Desktop/retail-data-analytics/Features-dataset.csv', stringsAsFactors = F)
sales <- read.csv('/Users/Natnicha/Desktop/retail-data-analytics/sales-dataset.csv', stringsAsFactors = F)
stores <- read.csv('/Users/Natnicha/Desktop/retail-data-analytics/stores-dataset.csv', stringsAsFactors = F)

1. [Objective of the Project](#head1)
2. [Data Exploration](#head2)
3. [Merg](#head3)
4. [Simple Linear Regression](#head4)
5. [Model Comparison](#head5)
6. [Multiple Linear Regression](#head6)
7. [Summary](#head8)
8. [Recommendation](#head9)

## <a id="head1"></a><font color='blue'> 1) Objective of the Project

Amazon is the world’s largest online retailer that offers a variety of product lines and services to customers. Amazon Prime is a subscription service that gives members access to a variety of features. Members receive benefits which include Amazon Prime Video, free 1 or 2-day shipping, unlimited reading, shopping exclusive deals on Amazon Prime Day, and a variety of other Amazon-specific services. Amazon Prime member is constantly growing in the United States because it is very convenient for members to purchase products from Amazon and gain access to all the services. 

The aim of this project is to develop the most comprehensive analysis of the shoppers using the historical sales data from the Retail Data Analytics dataset from Kaggle. By analyzing this dataset, Amazon will be able to understand its target market better and make smarter decisions in marketing.

The linear regression model will pool data together to help Amazon make informed decisions. The model will show Amazon how well each factor can predict the weekly sales, so that Amazon can decide on marketing strategies, and also maximize its marketing resources to improve the sales. This project could help Amazon acquire new customers and increase the sales and profit by analyzing the important factors that influence consumer spending. 


## <a id="head2"></a><font color='blue'> 2) Data Exploration

<b> 2.1) Print out the first 10 rows of dataset.</b>

In [2]:
head(features, n=10)

Store Date       Temperature Fuel_Price MarkDown1 MarkDown2 MarkDown3
1  1     05/02/2010 42.31       2.572      NA        NA        NA       
2  1     12/02/2010 38.51       2.548      NA        NA        NA       
3  1     19/02/2010 39.93       2.514      NA        NA        NA       
4  1     26/02/2010 46.63       2.561      NA        NA        NA       
5  1     05/03/2010 46.50       2.625      NA        NA        NA       
6  1     12/03/2010 57.79       2.667      NA        NA        NA       
7  1     19/03/2010 54.58       2.720      NA        NA        NA       
8  1     26/03/2010 51.45       2.732      NA        NA        NA       
9  1     02/04/2010 62.27       2.719      NA        NA        NA       
10 1     09/04/2010 65.86       2.770      NA        NA        NA       
   MarkDown4 MarkDown5 CPI      Unemployment IsHoliday
1  NA        NA        211.0964 8.106        FALSE    
2  NA        NA        211.2422 8.106         TRUE    
3  NA        NA        211.2891 8.106        FALSE    
4  NA        NA        211.3196 8.106        FALSE    
5  NA        NA        211.3501 8.106        FALSE    
6  NA        NA        211.3806 8.106        FALSE    
7  NA        NA        211.2156 8.106        FALSE    
8  NA        NA        211.0180 8.106        FALSE    
9  NA        NA        210.8204 7.808        FALSE    
10 NA        NA        210.6229 7.808        FALSE

In [3]:
head(sales, n=10)

Store Dept Date       Weekly_Sales IsHoliday
1  1     1    05/02/2010 24924.50     FALSE    
2  1     1    12/02/2010 46039.49      TRUE    
3  1     1    19/02/2010 41595.55     FALSE    
4  1     1    26/02/2010 19403.54     FALSE    
5  1     1    05/03/2010 21827.90     FALSE    
6  1     1    12/03/2010 21043.39     FALSE    
7  1     1    19/03/2010 22136.64     FALSE    
8  1     1    26/03/2010 26229.21     FALSE    
9  1     1    02/04/2010 57258.43     FALSE    
10 1     1    09/04/2010 42960.91     FALSE

In [4]:
head(stores, n=10)

Store Type Size  
1   1    A    151315
2   2    A    202307
3   3    B     37392
4   4    A    205863
5   5    B     34875
6   6    A    202505
7   7    B     70713
8   8    A    155078
9   9    B    125833
10 10    B    126512

<b> 2.2) Explore columns in the data set to see what type of columns they are. </b>

In [5]:
str(features)

'data.frame':	8190 obs. of  12 variables:
 $ Store       : int  1 1 1 1 1 1 1 1 1 1 ...
 $ Date        : chr  "05/02/2010" "12/02/2010" "19/02/2010" "26/02/2010" ...
 $ Temperature : num  42.3 38.5 39.9 46.6 46.5 ...
 $ Fuel_Price  : num  2.57 2.55 2.51 2.56 2.62 ...
 $ MarkDown1   : num  NA NA NA NA NA NA NA NA NA NA ...
 $ MarkDown2   : num  NA NA NA NA NA NA NA NA NA NA ...
 $ MarkDown3   : num  NA NA NA NA NA NA NA NA NA NA ...
 $ MarkDown4   : num  NA NA NA NA NA NA NA NA NA NA ...
 $ MarkDown5   : num  NA NA NA NA NA NA NA NA NA NA ...
 $ CPI         : num  211 211 211 211 211 ...
 $ Unemployment: num  8.11 8.11 8.11 8.11 8.11 ...
 $ IsHoliday   : logi  FALSE TRUE FALSE FALSE FALSE FALSE ...


In [6]:
str(sales)

'data.frame':	421570 obs. of  5 variables:
 $ Store       : int  1 1 1 1 1 1 1 1 1 1 ...
 $ Dept        : int  1 1 1 1 1 1 1 1 1 1 ...
 $ Date        : chr  "05/02/2010" "12/02/2010" "19/02/2010" "26/02/2010" ...
 $ Weekly_Sales: num  24924 46039 41596 19404 21828 ...
 $ IsHoliday   : logi  FALSE TRUE FALSE FALSE FALSE FALSE ...


In [7]:
str(stores)

'data.frame':	45 obs. of  3 variables:
 $ Store: int  1 2 3 4 5 6 7 8 9 10 ...
 $ Type : chr  "A" "A" "B" "A" ...
 $ Size : int  151315 202307 37392 205863 34875 202505 70713 155078 125833 126512 ...


<b>2.3) Explore the total number of rows and columns in each dataset </b>

In [8]:
cat("Features's total rows = ", nrow(features), "\n")
cat("Features's total columns = ", ncol(features))

Features's total rows =  8190 
Features's total columns =  12

In [9]:
cat("Sales' total rows = ", nrow(sales), "\n")
cat("Sales' total columns = ", ncol(sales))

Sales' total rows =  421570 
Sales' total columns =  5

In [10]:
cat("Stores' total rows = ", nrow(stores), "\n")
cat("Stores' total columns = ", ncol(stores))

Stores' total rows =  45 
Stores' total columns =  3

<b> 2.4) Summarize and provide statistic information about dataset </b>

In [11]:
summary(features)

     Store        Date            Temperature       Fuel_Price   
 Min.   : 1   Length:8190        Min.   : -7.29   Min.   :2.472  
 1st Qu.:12   Class :character   1st Qu.: 45.90   1st Qu.:3.041  
 Median :23   Mode  :character   Median : 60.71   Median :3.513  
 Mean   :23                      Mean   : 59.36   Mean   :3.406  
 3rd Qu.:34                      3rd Qu.: 73.88   3rd Qu.:3.743  
 Max.   :45                      Max.   :101.95   Max.   :4.468  
                                                                 
   MarkDown1        MarkDown2           MarkDown3           MarkDown4       
 Min.   : -2781   Min.   :  -265.76   Min.   :  -179.26   Min.   :    0.22  
 1st Qu.:  1578   1st Qu.:    68.88   1st Qu.:     6.60   1st Qu.:  304.69  
 Median :  4744   Median :   364.57   Median :    36.26   Median : 1176.42  
 Mean   :  7032   Mean   :  3384.18   Mean   :  1760.10   Mean   : 3292.94  
 3rd Qu.:  8923   3rd Qu.:  2153.35   3rd Qu.:   163.15   3rd Qu.: 3310.01  
 Max.   :1

In [12]:
summary(sales)

     Store           Dept           Date            Weekly_Sales   
 Min.   : 1.0   Min.   : 1.00   Length:421570      Min.   : -4989  
 1st Qu.:11.0   1st Qu.:18.00   Class :character   1st Qu.:  2080  
 Median :22.0   Median :37.00   Mode  :character   Median :  7612  
 Mean   :22.2   Mean   :44.26                      Mean   : 15981  
 3rd Qu.:33.0   3rd Qu.:74.00                      3rd Qu.: 20206  
 Max.   :45.0   Max.   :99.00                      Max.   :693099  
 IsHoliday      
 Mode :logical  
 FALSE:391909   
 TRUE :29661    
                
                
                

In [13]:
summary(stores)

     Store        Type                Size       
 Min.   : 1   Length:45          Min.   : 34875  
 1st Qu.:12   Class :character   1st Qu.: 70713  
 Median :23   Mode  :character   Median :126512  
 Mean   :23                      Mean   :130288  
 3rd Qu.:34                      3rd Qu.:202307  
 Max.   :45                      Max.   :219622  

## <a id="head3"></a><font color='blue'> 3) Merge

In [14]:
fs <- merge(features, sales, by = c("Store",  "Date", "IsHoliday"))


In [15]:
sales_all <- merge(fs, stores, by = "Store")

In [16]:
nrow(sales_all)
ncol(sales_all)

[1] 421570

[1] 16

In [17]:
head(sales_all, n=10)

Store Date       IsHoliday Temperature Fuel_Price MarkDown1 MarkDown2
1  1     01/04/2011 FALSE     59.17       3.524      NA        NA       
2  1     01/04/2011 FALSE     59.17       3.524      NA        NA       
3  1     01/04/2011 FALSE     59.17       3.524      NA        NA       
4  1     01/04/2011 FALSE     59.17       3.524      NA        NA       
5  1     01/04/2011 FALSE     59.17       3.524      NA        NA       
6  1     01/04/2011 FALSE     59.17       3.524      NA        NA       
7  1     01/04/2011 FALSE     59.17       3.524      NA        NA       
8  1     01/04/2011 FALSE     59.17       3.524      NA        NA       
9  1     01/04/2011 FALSE     59.17       3.524      NA        NA       
10 1     01/04/2011 FALSE     59.17       3.524      NA        NA       
   MarkDown3 MarkDown4 MarkDown5 CPI      Unemployment Dept Weekly_Sales Type
1  NA        NA        NA        214.8372 7.682        49   13167.85     A   
2  NA        NA        NA        214.8372 7.682        26    5946.53     A   
3  NA        NA        NA        214.8372 7.682        81   28545.23     A   
4  NA        NA        NA        214.8372 7.682        34    9949.54     A   
5  NA        NA        NA        214.8372 7.682        59     316.86     A   
6  NA        NA        NA        214.8372 7.682        30    3897.48     A   
7  NA        NA        NA        214.8372 7.682         7   20144.71     A   
8  NA        NA        NA        214.8372 7.682        85    3208.74     A   
9  NA        NA        NA        214.8372 7.682         8   35319.05     A   
10 NA        NA        NA        214.8372 7.682        28     602.63     A   
   Size  
1  151315
2  151315
3  151315
4  151315
5  151315
6  151315
7  151315
8  151315
9  151315
10 151315

## <a id="head4"></a><font color='blue'>4) Simple Linear Regression

<b>4.1) Relationship between weekly sales and temperature </b>

In [18]:
sales.temp <- lm(Weekly_Sales ~ Temperature, data = sales_all)
summary(sales.temp)


Call:
lm(formula = Weekly_Sales ~ Temperature, data = sales_all)

Residuals:
   Min     1Q Median     3Q    Max 
-20937 -13906  -8368   4224 677105 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 16152.325    119.184 135.524   <2e-16 ***
Temperature    -2.847      1.896  -1.501    0.133    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 22710 on 421568 degrees of freedom
Multiple R-squared:  5.347e-06,	Adjusted R-squared:  2.975e-06 
F-statistic: 2.254 on 1 and 421568 DF,  p-value: 0.1332


The Adjusted R-squared value of 2.975e-06 indicates that the model explains over 0 percent of the variance in the data. Our model doesn't fit the data. p-value > 0.05 (not significant) We don't have enough evidence to claim that there is a relationship between weekly sales and temperature. Therefore, we can conclude that temperature should not be used to predict weekly sales

<b>4.2) Relationship between weekly sales and CPI</b>

In [19]:
sales.cpi <- lm(Weekly_Sales ~ CPI, data = sales_all)
summary(sales.cpi)


Call:
lm(formula = Weekly_Sales ~ CPI, data = sales_all)

Residuals:
   Min     1Q Median     3Q    Max 
-21515 -13842  -8358   4224 676578 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept) 18058.5772   156.8406  115.14   <2e-16 ***
CPI           -12.1337     0.8931  -13.59   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 22710 on 421568 degrees of freedom
Multiple R-squared:  0.0004377,	Adjusted R-squared:  0.0004353 
F-statistic: 184.6 on 1 and 421568 DF,  p-value: < 2.2e-16


The Adjusted R-squared value of 0.000435 indicates that the model explains 0.05 percent of the variance in the data; approximately 6% of variation in weekly sales can be explained by CPI.  p-value < 0.05 (significant) We have strong evidence that there is, in fact, a relationship between weekly sales and CPI. Eventhough R-squared value is low, we have statistically significant independent variable. CPI is correlated with weekly sales, but it doesn't explain much of the variability in the weekly sales. CPI is negatively related to weekly sales due to its negative slope. Therefore, we can conclude that CPI can be used to predict weekly sales

<b>4.3) Relationship between weekly sales and fuel price</b>

In [20]:
sales.fuel <- lm(Weekly_Sales ~ Fuel_Price, data = sales_all)
summary(sales.fuel)


Call:
lm(formula = Weekly_Sales ~ Fuel_Price, data = sales_all)

Residuals:
   Min     1Q Median     3Q    Max 
-20972 -13901  -8368   4225 677117 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 16001.285    258.779  61.834   <2e-16 ***
Fuel_Price     -5.958     76.287  -0.078    0.938    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 22710 on 421568 degrees of freedom
Multiple R-squared:  1.447e-08,	Adjusted R-squared:  -2.358e-06 
F-statistic: 0.006101 on 1 and 421568 DF,  p-value: 0.9377


The Adjusted R-squared value of -2.358e-06 (negative) indicates that our model doesn't follow the trend of the data. p-value > 0.05 (not significant) We don't have enough evidence to claim that there is a relationship between weekly sales and fuel price. Therefore, we can conclude that fuel price shouldn't be used to predict weekly sales.

<b>4.4) Relationship between weekly sales and unemployment rate</b>

In [21]:
sales.unemp <- lm(Weekly_Sales ~ Unemployment, data = sales_all)
summary(sales.unemp)


Call:
lm(formula = Weekly_Sales ~ Unemployment, data = sales_all)

Residuals:
   Min     1Q Median     3Q    Max 
-19677 -13872  -8376   4244 677447 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  18490.70     153.42   120.5   <2e-16 ***
Unemployment  -315.25      18.77   -16.8   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 22700 on 421568 degrees of freedom
Multiple R-squared:  0.0006689,	Adjusted R-squared:  0.0006666 
F-statistic: 282.2 on 1 and 421568 DF,  p-value: < 2.2e-16


The Adjusted R-squared value of 0.0006689 indicates that the model explains over 0.07 percent of the variance in the data; approximately 0.07% of variation in weekly sales can be explained by unemployment rate. p-value < 0.05 (significant) We have strong evidence that there is, in fact, a relationship between weekly sales and unemployment rate.  Eventhough R-squared value is low, we have statistically significant predictors. Unemployment rate is correlated with weekly sales, but it doesn't explain much of the variability in the weekly sales. Unemployment rate is negatively related to weekly sales due to its negative slope. Therefore, we can conclude that unemployment rate can be used to predict weekly sales.

<b>4.5) Relationship between weekly sales and store size</b>

In [22]:
sales.size <- lm(Weekly_Sales ~ Size, data = sales_all)
summary(sales.size)


Call:
lm(formula = Weekly_Sales ~ Size, data = sales_all)

Residuals:
   Min     1Q Median     3Q    Max 
-27288 -12872  -6620   4283 678046 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 3.565e+03  8.328e+01   42.81   <2e-16 ***
Size        9.081e-02  5.563e-04  163.24   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 22030 on 421568 degrees of freedom
Multiple R-squared:  0.05945,	Adjusted R-squared:  0.05945 
F-statistic: 2.665e+04 on 1 and 421568 DF,  p-value: < 2.2e-16


The Adjusted R-squared value of 0.0595 indicates that the model explains over 6 percent of the variance in the data; approximately 6% of variation in weekly sales can be explained by store size.
p-value < 0.05 (significant) We have strong evidence that there is, in fact, a relationship between weekly sales and  store size. Eventhough R-squared value is low, we have statistically significant predictor. Store size is correlated with weekly sales, but it doesn't explian much of the variability in the weekly sales. Store size is positively related to weekly sales due to its positive slope. Therefore, we can conclude that strore size can be used to predict our weekly sales.

## <a id="head5"></a><font color='blue'>5) Model Comparison

A predictor that has a low p-value is likely to be a meaningful addition to our model because changes in the predictor's value are related to changes in the response variable. Moreover, the higher r-square is, the better to explain changes in the outcome variable. 
After Comparing each of the models in 4), I decide that the <b>sales.size model</b> (relationship between weekly sales and store size) is better than the rest. This model has the least p-value(2.2e-16) and the most R-squared(0.06). This model has independent variables that are statistically significant but a low R-squared value. This combination indicates that store size are correlated with weekly sales, but it doesn't explin much of the variability in the weekly sales. 


Therefore, the best model so far is `Weekly_Sales = (3.565e+03) + (9.081e-02) * Size`

## <a id="head6"></a><font color='blue'>6) Multiple Linear Regression

<b> 6.1) Relationship between weekly sale, unemployment rate, and MarkDown1</b>

In [25]:
m1 <- lm(Weekly_Sales ~ Unemployment + MarkDown1 , data = sales_all)
summary(m1)


Call:
lm(formula = Weekly_Sales ~ Unemployment + MarkDown1, data = sales_all)

Residuals:
   Min     1Q Median     3Q    Max 
-36045 -13689  -8600   4333 635659 

Coefficients:
               Estimate Std. Error t value Pr(>|t|)    
(Intercept)   1.808e+04  2.656e+02   68.08   <2e-16 ***
Unemployment -4.913e+02  3.460e+01  -14.20   <2e-16 ***
MarkDown1     2.443e-01  7.206e-03   33.91   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 23160 on 150678 degrees of freedom
  (270889 observations deleted due to missingness)
Multiple R-squared:  0.008595,	Adjusted R-squared:  0.008581 
F-statistic: 653.1 on 2 and 150678 DF,  p-value: < 2.2e-16


In this case, we can see the R-squared of 0.0086, approximately 0.86% of variation in weekly sales can be explained by unemployment rate and MarkDown1. The p-value for over all model is 2.2e-16. This is substantially lower than 0.05, we generally reject the null hypothesis. In other words, with a p-value less than 0.05, we have strong evidence that there is, in fact, a relationship between the unemployment rate, MarkDown1 and weekly sales. 

<b> 6.2) Relationship between weekly sales, Unemployement rate, MarkDown1 and temperature </b>

In [26]:
m2 <- lm(Weekly_Sales ~ Unemployment + MarkDown1 + Temperature , data = sales_all)
summary(m2)


Call:
lm(formula = Weekly_Sales ~ Unemployment + MarkDown1 + Temperature, 
    data = sales_all)

Residuals:
   Min     1Q Median     3Q    Max 
-36097 -13690  -8597   4345 635604 

Coefficients:
               Estimate Std. Error t value Pr(>|t|)    
(Intercept)   1.834e+04  3.326e+02   55.13   <2e-16 ***
Unemployment -4.905e+02  3.460e+01  -14.18   <2e-16 ***
MarkDown1     2.440e-01  7.213e-03   33.83   <2e-16 ***
Temperature  -4.338e+00  3.389e+00   -1.28    0.201    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 23160 on 150677 degrees of freedom
  (270889 observations deleted due to missingness)
Multiple R-squared:  0.008605,	Adjusted R-squared:  0.008586 
F-statistic:   436 on 3 and 150677 DF,  p-value: < 2.2e-16


In this case, the p-value for over all model is 2.2e-16. This is substantially lower than 0.05, we generally reject the null hypothesis. In other words, with a p-value less than 0.05, we have strong evidence that there is, in fact, a relationship between the three predictors and weekly sales. Perhaps the temperature variable is unnecessary in the model given its high p-value of 0.201 which indicates that the temperature doesn't help explain the model well.



Next, I use `anova( )` to compare m1 and m2. This will tell us whether adding temperature on top of unemployment rate and MarkDown1 further improves the model.

In [27]:
anova(m1,m2)

Res.Df RSS          Df Sum of Sq F        Pr(>F)   
1 150678 8.084966e+13 NA        NA       NA        NA
2 150677 8.084878e+13  1 879069333 1.638312 0.2005592

The result shows a significant result (p = 0.2006). Thus, we shiould reject m2 and stick with model 1

Thus, the best model so far is `Weekly_Sales = 1.808e+04 + (-4.913e+02) * Unnemployment + (2.443e-01) * MarkDown1`

<b> 6.3) Relationship between weekly sales, store size, and Markdown3</b>

In [28]:
m3 <- lm(Weekly_Sales ~ Size + MarkDown3, data = sales_all)
summary(m3)



Call:
lm(formula = Weekly_Sales ~ Size + MarkDown3, data = sales_all)

Residuals:
   Min     1Q Median     3Q    Max 
-36962 -13510  -6794   4595 629609 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 3.483e+03  1.616e+02   21.55   <2e-16 ***
Size        9.236e-02  1.054e-03   87.59   <2e-16 ***
MarkDown3   1.213e-01  6.484e-03   18.70   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 23080 on 137088 degrees of freedom
  (284479 observations deleted due to missingness)
Multiple R-squared:  0.05645,	Adjusted R-squared:  0.05644 
F-statistic:  4101 on 2 and 137088 DF,  p-value: < 2.2e-16


In this case, we can see the R-squared of 0.056, approximately 5.6% of variation in weekly sales can be explained by store size and MarkDown3. The p-value for over all model is 2.2e-16. This is substantially lower than 0.05, we generally reject the null hypothesis. In other words, with a p-value less than 0.05, we have strong evidence that there is, in fact, a relationship between the store size, MarkDown3 and weekly sales. 

<b> 6.4) Relationship between weekly sales, store size, Markdown3, and CPI</b>

In [29]:
m4 <- lm(Weekly_Sales ~ Size + MarkDown3 + CPI, data = sales_all)
summary(m4)


Call:
lm(formula = Weekly_Sales ~ Size + MarkDown3 + CPI, data = sales_all)

Residuals:
   Min     1Q Median     3Q    Max 
-37352 -13507  -6671   4629 629291 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept)  5.433e+03  3.147e+02  17.266  < 2e-16 ***
Size         9.244e-02  1.054e-03  87.683  < 2e-16 ***
MarkDown3    1.202e-01  6.485e-03  18.529  < 2e-16 ***
CPI         -1.122e+01  1.553e+00  -7.223 5.11e-13 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 23070 on 137087 degrees of freedom
  (284479 observations deleted due to missingness)
Multiple R-squared:  0.05681,	Adjusted R-squared:  0.05679 
F-statistic:  2752 on 3 and 137087 DF,  p-value: < 2.2e-16


Here, we can see the R-squared of 0.057, approximately 5.7% of variation in weekly sales can be explained by store size, MarkDown3, and CPI. The p-value for over all model is 2.2e-16. This is substantially lower than 0.05, we generally reject the null hypothesis. In other words, with a p-value less than 0.05, we have strong evidence that there is, in fact, a relationship between the store size, MarkDown3, CPI and weekly sales. 

Next, I use `anova( )` to compare m3 and m4. This will tell us whether adding CPI on top of Size and MarkDown3 further improves the model.

In [30]:
anova(m3,m4)

Res.Df RSS          Df Sum of Sq   F        Pr(>F)      
1 137088 7.299511e+13 NA          NA       NA           NA
2 137087 7.296734e+13  1 27770353774 52.17341 5.107203e-13

The result shows a significant result (p = 5.107203e-13). Thus, adding CPI further improves our model.

Thus, the best model so far is `Weekly_Sales = (5.433e+03) + (9.244e-02) * Size + (1.202e-01) * MarkDown3 + (-1.122e+01) * CPI`

By comparing the multiple regression result with the simple regression result, I found that multiple regression model has more power than simple regression model in predicting weekly sales, but we have to keep in mind that we should  add only the factor will further improve our model.

## <a id="head7"></a><font color='blue'>7) Summary

After analyzing the dataset, I found that p-value of temperature and fuel price are not significant; changes in temperature and fuel price are not associated with changes in weekly sales. Since we don't have enough evidence to claim that there is a relationship between weekly sales and temperature, and weekly sales and fuel price, they should not be used to predict weekly sales.  

Moreover, I found that p-value of CPI, unemployment rate, and store size are significant; changes in these preditor's value are related to changes in weekly sales. Since the slope of unemployment rate and CPI is negative. It indicates that weekly sales and CPI are negatively related. For example, when unemployment rate increases, weekly sales decreases,and when unemployment rate decreases, weekly sales increases. Conversely, since the slope of store size is positive, store size and weekly sales are positively related; that is when store size increases, weekly sales increases,and when store size decreases, weekly sales decreases.

By looking at the multiple linear regression models, although, we can predict dependent variable better by adding multiple factors to the model, we need to be careful and check a significant result every time we add new factor to our model because that factor might cause less accurate in predicting. 

## <a id="head8"></a><font color='blue'>8) Recommendations

Base on my analysis, I would recommend that Amazon should pay more attention on the value of CPI, unemployment rate and store size before launching any products or services. If Amazon want people to spend money on their products or services, Amazon need to make an investment at the right time. For example, if Amazon launch new products while the unemployment rate is high, a lot of people might be interested but could not afford it because they are unemployed; Amazon's sales might decrease and lose money eventually. 

As we can see that it is very important to know which factors may have an impact on customer spending. Amazon need to study those factors that are significant in order to understand the market to better target shopper. Finally, Amazon can make informed decisions on marketing strategies to improve the sales and profit.
